verticapy/plotting/_plotly/outliers.py

```diff
@@ -0,0 +1,175 @@
+"""
+(c)  Copyright  [2018-2023]  OpenText  or one of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+from typing import Literal, Optional
+from verticapy._typing import ArrayLike
+
+import pandas as pd
+import numpy as np
+import plotly.express as px
+import plotly.graph_objects as go
+from plotly.graph_objs._figure import Figure
+
+from verticapy.plotting._plotly.base import PlotlyBase
+
+
+class OutliersPlot(PlotlyBase):
+    # Properties.
+
+    @property
+    def _category(self) -> Literal["plot"]:
+        return "plot"
+
+    @property
+    def _kind(self) -> Literal["outliers"]:
+        return "outliers"
+
+    @property
+    def _compute_method(self) -> Literal["outliers"]:
+        return "outliers"
+
+    @property
+    def _dimension_bounds(self) -> tuple[int, int]:
+        return (1, 2)
+
+    # Styling Methods.
+
+    def _init_style(self) -> None:
+        self.init_style = {"width": 500, "height": 500}
+        return None
+
+    # Draw.
+
+    def draw(
+        self,
+        fig: Optional[Figure] = None,
+        colorscale: Optional[ArrayLike] = None,
+        **style_kwargs,
+    ) -> Figure:
+        """
+        Draws an outliers contour plot using the Plotly API.
+        """
+        fig = self._get_fig(fig)
+        if len(self.layout["columns"]) == 1:
+            X1 = self.data["inliers"][:, 0].flatten().tolist()
+            X2 = self.data["outliers"][:, 0].flatten().tolist()
+            cat_X1 = ["inliers"] * len(X1)
+            cat_X2 = ["outliers"] * len(X2)
+            x_axis = ["age"] * (len(X1) + len(X2))
+            df = pd.DataFrame(
+                {
+                    self.layout["columns"][0]: X1 + X2,
+                    "category": cat_X1 + cat_X2,
+                    "x_axis": x_axis,
+                }
+            )
+            fig_scatter = px.strip(
```

Comment:
the scatter is the same 1D and 2D, you'll have a Matrix with the elements.

Reply:
If you look at the outliers plot of matplotlib (or even highchart), scatters calls do not change from one dimension to another so they should be drawn outside of your condition.

Reply:
![image](https://user-images.githubusercontent.com/46414488/230187858-c11414b8-ef78-4b9b-b31d-b61c068b475b.png)

This is because my 1d plot is a bit different. It is without the contour. I thought it was more cleaner. What do you think?

Reply:
good